code/__defines/_macros.dm

```diff
@@ -4,6 +4,12 @@
 #define list_find(L, needle, LIMITS...) L.Find(needle, LIMITS)
 #define hex2num(hex) text2num(hex, 16)
 #define num2hex(num, pad) num2text(num, pad, 16)
+
+/// Takes a datum as input, returns its ref string, or a cached version of it
+/// This allows us to cache \ref creation, which ensures it'll only ever happen once per datum, saving string tree time
+/// It is slightly less optimal then a []'d datum, but the cost is massively outweighed by the potential savings
+/// It will only work for datums mind, for datum reasons
+/// : because of the embedded typecheck
```

Comment:
```suggestion
/** 
  * Takes a datum as input, returns its ref string, or a cached version of it
  * 
  * This allows us to cache \ref creation, which ensures it'll only ever happen once per datum, saving string tree time
  * It is slightly less optimal then a []'d datum, but the cost is massively outweighed by the potential savings
  * It will only work for datums mind, for datum reasons
  * : because of the embedded typecheck
  */
```

Reply:
This would be a diff from the port's source without any benefit